### PR TITLE
sharing feature tests

### PR DESCRIPTION
**Pull request by SergioBertolinSG**
Added first test. 
Own remainder: As An method should be changed a little since using a different user than admin, the test fails.

### CODE REVIEW

File: build/integration/features/sharing-v1.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
+Feature: sharing
+  Background:
+    Given using api version "1"
+
+  Scenario: Create user user0
```
Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: this is no scenario in the feature sharing.

In a sharing scenario you simple define the user in a given statment

File: build/integration/features/sharing-v1.feature [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  Scenario: Create a group
+    Given As an "admin"
+    And group "sharing-group" does not exist
+    When sending "POST" to "/cloud/groups" with
+      | groupid | sharing-group |
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And group "sharing-group" exists
+
+  Scenario: Creating a new share
```
Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: this is the only true sceanrio - all other should be defined as a Given sentence

**Comment by SergioBertolinSG** [comment]
Please review @DeepDiver1975 @MorrisJobke

File: build/integration/features/sharing-v1.feature [hunk 3]
```diff
@@ -0,0 +1,79 @@
+Feature: sharing
+  Background:
+    Given using api version "1"
+
+  Scenario: Creating a new share with user
+    Given As an "admin"
```
Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: Create user is not a valid given statement.

```
Given user "user0" exists
And user "user1" exists
And as "user1"
When sending .....

File: build/integration/features/sharing-v1.feature [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    And Create user "user0"
+    And Create user "user1"
+    And As an "user0"
+    When sending "POST" to "/apps/files_sharing/api/v1/shares" with
+      | path | welcome.txt |
+      | shareWith | user1 |
+      | shareType | 0 |
+    Then the OCS status code should be "100"
+    And the HTTP status code should be "200"
+    And As an "admin"
```
Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: no cleanup in the assertion statements

File: build/integration/features/sharing-v1.feature (hunk 3 above)
Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: as an alternative a set of users and groups can be defined in the background

Comment (SergioBertolinSG) [inline build/integration/features/sharing-v1.feature]: The point is, every scenario needs some preconditions and postconditions to be executed, previously I used other scenarios to do it. Now using the given/then statements. Without this statements tests are not possible.

Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: @SergioBertolinSG it's only a different naming of the statements - the sentences are not to be understood as operations but as preconditions.

"A user exists" is a precondition - "Create user is not"

Comment (DeepDiver1975) [inline build/integration/features/sharing-v1.feature]: maybe this helps to get a common understanding https://github.com/cucumber/cucumber/wiki/Given-When-Then

Comment (SergioBertolinSG) [inline build/integration/features/sharing-v1.feature]: Ok, so I'll need to modify the userExists and similar functions to create what doesn't exist.

**Comment by SergioBertolinSG** [comment]
@DeepDiver1975 ping

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
@MorrisJobke mind having a look? THX

**Comment by MorrisJobke** [comment]
Looks good. :+1: 

Also the code is now a bit more structured to the actual logical functions, I like it :)

**Comment by DeepDiver1975** [comment]
> Also the code is now a bit more structured to the actual logical functions, I like it :)

now that we are on behat 3 we could also start to group features and use multiple contexts .... if needed

**Pull request merged**